### DISCUSSION TITLE
Roadmap for reformating to HyperSDK

### DISCUSSION THREAD

**Discussion by 0xHarrington** [ava-labs/hypersdk#11] [Chat]
Developers with existing bespoke AVMs may want to refactor their codebase to using HyperSDK. Documentation outlining distinct steps and suggestions for migrating to the hyper-opinionated HyperSDK would be helpful for guidance alongside the public release.

**Comment by patrick-ogrady** [comment]
I think we could provide high-level recommendations (I.e. turn your transactions into `Actions` and `Auth`) but it may be tough to provide specific guidance given the breadth of the AVM design space.

Maybe it would be more productive to collect a list of "gotchas"/"misconceptions most people have about using hypersdk" to serve this audience instead (not to mention it would help new builders too).
